crates/ruma-identity-service-api/src/association/bind_3pid/v2.rs

```diff
@@ -0,0 +1,78 @@
+//! [POST /_matrix/identity/v2/3pid/bind](https://matrix.org/docs/spec/identity_service/r0.3.0#post-matrix-identity-v2-3pid-bind)
+
+use std::collections::BTreeMap;
+
+use js_int::UInt;
+use ruma_api::ruma_api;
+use ruma_common::thirdparty::Medium;
+use ruma_identifiers::{ServerNameBox, ServerSigningKeyId, UserId};
+
+ruma_api! {
+    metadata: {
+        description: "Publish an association between a session and a Matrix user ID.",
+        method: POST,
+        name: "bind_3pid",
+        path: "/_matrix/identity/v2/3pid/bind",
+        rate_limited: false,
+        authentication: AccessToken,
+    }
+
+    request: {
+        /// The Session ID generated by the `requestToken` call.
+        pub sid: &'a str,
+
+        /// The client secret passed to the `requestToken` call.
+        pub client_secret: &'a str,
+
+        /// The Matrix user ID to associate with the 3PIDs.
+        pub mxid: &'a UserId,
+    }
+
+    response: {
+        /// The 3PID address of the user being looked up.
+        pub address: String,
+
+        /// The medium type of the 3PID.
+        pub medium: Medium,
+
+        /// The Matrix user ID associated with the 3PID.
+        pub mxid: UserId,
+
+        /// A UNIX timestamp before which the association is not known to be valid.
+        pub not_before: UInt,
```

Comment:
For time-based fields, we use the types from `ruma_common::time`. From the examples given in the spec, and the note [here](https://matrix.org/docs/spec/#timestamps), this looks like it represents milliseconds.  can you use the `ruma_common::time::MillisecondsSinceUnixEpoch` struct? Same with the other two below.

(Note to self: add references to these structs to the contributors guide.)